File: pyg_shop_web/src/main/resources/static/js/service/typeTemplateService.js
//服务层
app.service('typeTemplateService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../typeTemplate/findAll');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../typeTemplate?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../typeTemplate/'+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../typeTemplate',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.put('../typeTemplate',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.delete('../typeTemplate/'+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../typeTemplate/search?page='+page+"&rows="+rows, searchEntity);
	}

	// 根据模板id获取该模板的所有规格和规格选项
    this.findSpecList = function (id) {
		// alert(111);
        return $http.get("../typeTemplate/findSpecList/" + id);
    }
});

File: pyg_sellergoods_interface/src/main/java/cn/net/zhipeng/sellergoods/service/TypeTemplateService.java
package cn.net.zhipeng.sellergoods.service;


import cn.net.zhipeng.entity.PageResult;
import cn.net.zhipeng.pojo.TypeTemplate;

import java.util.List;
import java.util.Map;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface TypeTemplateService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TypeTemplate> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TypeTemplate typeTemplate);
	
	
	/**
	 * 修改
	 */
	public void update(TypeTemplate typeTemplate);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TypeTemplate findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TypeTemplate typeTemplate, int pageNum, int pageSize);

	/**
	 * 返回规格&规格下的所有选项
	 * @param id
	 * @return
	 */
	public List<Map> findSpecList(Long id);
}
